shell/platform/fuchsia/flutter/tests/fake_flatland_unittests.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "fakes/scenic/fake_flatland.h"
+
+#include <lib/async-testing/test_loop.h>
+#include <lib/async/dispatcher.h>
+
+#include <string>
+
+#include "flutter/fml/logging.h"
+#include "fuchsia/ui/composition/cpp/fidl.h"
+#include "gtest/gtest.h"
+
+namespace flutter_runner::testing {
+namespace {
+
+std::string GetCurrentTestName() {
+  return ::testing::UnitTest::GetInstance()->current_test_info()->name();
+}
+
+}  // namespace
+
+class FakeFlatlandTest : public ::testing::Test {
+ protected:
+  FakeFlatlandTest() : flatland_subloop_(loop_.StartNewLoop()) {}
+  ~FakeFlatlandTest() override = default;
+
+  async::TestLoop& loop() { return loop_; }
+
+  FakeFlatland& fake_flatland() { return fake_flatland_; }
+
+  fuchsia::ui::composition::FlatlandPtr CreateFlatland() {
+    FML_CHECK(!fake_flatland_.is_bound());
+
+    auto flatland_handle = fake_flatland_.Bind(flatland_subloop_->dispatcher());
+    return flatland_handle.Bind();
+  }
+
+ private:
+  async::TestLoop loop_;  // Must come before FIDL bindings.
+
+  std::unique_ptr<async::LoopInterface> flatland_subloop_;
```

Comment:
Perhaps describe why we need a sub-loop. Don't think it's obvious.

Reply:
Done